### PR TITLE
Add text_box_search, begin linting

### PR DESCRIPTION
**Pull request by NickBethune**

### CODE REVIEW

**Comment by NickBethune** [comment]
@justinnhli @crzeller11

File: frontend_search_test.py [hunk 1]
```diff
@@ -1,27 +1,36 @@
 from flask import Flask, render_template, flash, request
-from wtforms import Form, TextField, TextAreaField, validators, StringField, SubmitField
-from search import prepare_tree, search, find_artic,get_measure
+from wtforms import Form, TextField, validators
 from lxml import etree
+from search import prepare_tree, search, find_artic, get_measure
 
 """Turn user submitted MEI code snippet into xml file."""
```
Comment (justinnhli) [inline frontend_search_test.py]: Move this string inside the function for it to count as a docstring:
```python
 def transform_xml(xml_code):
    """Turn user submitted MEI code snippet into xml file."""
    xml = xml_code.encode()
    ...
```

File: frontend_search_test.py [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
+
+
 def transform_xml(xml_code):
     xml = xml_code.encode()
-    f = open('upload.xml', 'wb')
-    f.write(xml)
-    f.close()
+    file = open('upload.xml', 'wb')
```
Comment (justinnhli) [inline frontend_search_test.py]: use context manager syntax instead:

```python
with open('upload.xml', 'wb') as file_descriptor:
    file_descriptor.write(xml)
```

Comment (justinnhli) [reply frontend_search_test.py]: Also, is the 'b' in 'wb' necessary? xml should be mostly text, so it's unclear you need to write binary.

File: frontend_search_test.py (hunk 1 above)
Comment (justinnhli) [reply frontend_search_test.py]: This applies to all subsequent docstrings

File: search.py [hunk 3]
```diff
@@ -135,34 +168,34 @@ def notes_on_beam(tree):
...
+                beam_children += '0'
 
             # else if the child is a chord, add a list of notes to the list
             elif child.tag == '{http://www.music-encoding.org/ns/mei}chord':
                 notes = child.getchildren()
-                ls.append([note.attrib['pname'] for note in notes if note.tag ==
-                           '{http://www.music-encoding.org/ns/mei}note'])
-        beam_notes_list.append(ls)
+                beam_children.append([note.attrib['pname'] for note in notes if note.tag ==
+                                      '{http://www.music-encoding.org/ns/mei}note'])
```
Comment (justinnhli) [inline search.py]: I personally prefer the following, but whether you change it is up to you:

```python
beam_children.append([
    note.attrib['pname'] for note in notes
    if note.tag == '{http://www.music-encoding.org/ns/mei}note'
])
```

File: search.py [hunk 4]
```diff
@@ -236,7 +293,7 @@ def search(input_root, data_tree):
     data_list = root_to_list(data_tree.getroot())
     measure_match_list = []
 
-    # iterate over data MEI file
+    # iterate over data MEI file ele
```
Comment (justinnhli) [inline search.py]: typo?

File: frontend_search_test.py [hunk 5]
```diff
@@ -1,29 +1,35 @@
...
 
 """Construct the submission box and taxonomies"""
+
+
 class SnipUpload(Form):
     name = TextField('Submit your MEI Code Snippets:', validators=[validators.required()])
 
-
-"""Snippet submission"""
+x
```
Comment (justinnhli) [inline frontend_search_test.py]: typo

**Review by justinnhli — APPROVED** [review]
fix the typo and I'll merge

**Pull request merged**